neptune/alpha/internal/utils/json_file_splitter.py

```diff
@@ -43,14 +44,20 @@ def get(self) -> Optional[dict]:
                 return None
 
     def _read_data(self):
+        if self._start_pos > 0:
+            new_buffer = StringIO()
+            new_buffer.write(self._buffer.getvalue()[self._start_pos:])
```

Comment:
I don't know why are you reading every time whole buffer into str and then select part of it. You are reading here into str even this parts that you already read. This is very suboptimal.